### ISSUE TITLE
[feature] js_func (for javascript) equivalent of py_func

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#18280]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
n/a
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
iOS, Android
- **TensorFlow installed from (source or binary)**:
n/a
- **TensorFlow version (use command below)**:
n/a
- **Python version**: 
n/a
- **Bazel version (if compiling from source)**:
n/a
- **GCC/Compiler version (if compiling from source)**:
n/a
- **CUDA/cuDNN version**:
n/a
- **GPU model and memory**:
n/a
- **Exact command to reproduce**:
n/a

### Describe the problem
I suggest adding a `js_func` that allows defining nodes in javascript. This would be similar to the existing `py_func`. There are many systems (ie iOS and Android) that have JS runtimes but do not have python runtimes.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by dsmilkov** [comment]
Hi @cancan101, the body of the py_func will not be serialized in a GraphDef. Therefore, you should not use this function if you need to serialize your model and restore it in a different environment.

Also, if you are using TensorFlow.js for development, and are thinking about this in a JS-only context, would you mind re-posting this issue to the [tensorflow/tfjs issues](https://github.com/tensorflow/tfjs/issues)?

Thanks!

**Comment by cancan101** [comment]
Interesting to know that the body is not serialized; I had not realized that. I was not thinking of js_func in the context of tfjs. Rather I was thinking of loading a graph on iOS or Android and using the js_func to supply user created functions.

**Comment by reedwm** [comment]
/CC @andrewharp can you take a look since this is desired in mobile?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 123 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 139 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 154 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 169 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 184 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 199 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 214 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrewharp: It has been 229 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tilakrayal** [comment]
@cancan101,
Please let me know if my understanding is correct here. Looks like this isssue is more related to TensorFlow JavaScript, if you are using TensorFlow.js for development and thinking about this in a Javascript context, could you please try to post this issue to the [tensorflow/tfjs issues](https://github.com/tensorflow/tfjs/issues)

Thank you!

**Comment by cancan101** [comment]
No, this is not a tfjs issue. This is about the ability to have a node that can run javascript, similar to the [py_func](https://www.tensorflow.org/api_docs/python/tf/py_function) node that runs python code.

**Comment by Venkat6871** [comment]
Hi @cancan101 ,
Thank you so much for filing this Feature request! We truly appreciate your support. At the moment, this request is not prioritized and mark it for community contributions. If you would like to contribute on a fix, please feel free to fork the repository and submit a pull request in accordance with the TensorFlow [CONTRIBUTION GUIDELINES](https://www.tensorflow.org/community/contribute/code) and link this issue in your PR to ensure the maintainers can easily review and collaborate.To avoid duplicate work, please leave a comment here if you decide to take it on!
Thank you again for your interest and we look forward to seeing your contributions!

**Comment by Venkat6871** [comment]
Hi @cancan101 ,

We are closing this issue due to inactivity. If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it.

Thank you for your understanding.
